lib/manageiq/providers/openstack/legacy/openstack_handle/handle.rb

```diff
@@ -80,7 +80,7 @@ def self.raw_connect(username, password, auth_url, service = "Compute", extra_op
       else
         Fog.const_get(service).new(opts)
       end
-    rescue Fog::OpenStack::Auth::Catalog::ServiceTypeError
+    rescue Fog::OpenStack::Auth::Catalog::ServiceTypeError, Fog::Service::NotFound
```

Comment:
I was also seeing a lot of "nfv service not available" exceptions even though we have a conditional on the service being non-nil.  This was because `detect_service` here was only catching `MiqException::ServiceNotAvailable` from down here https://github.com/ManageIQ/manageiq-providers-openstack/pull/701/files#diff-8b5a0f2181b4ddbbe590078b4a29f5bf5a4f763b151dd02f4951102497ee2af9R86 but the actual exception was `Fog::Service::NotFound`

Reply:
Looks like I was fixing the same issue as @aufi was here https://github.com/ManageIQ/manageiq-providers-openstack/pull/702